### PR TITLE
Support partition key

### PR DESCRIPTION
**Pull request by xige-16**
issue: https://github.com/milvus-io/milvus/issues/23553

Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

**Comment by xige-16** [comment]
rerun UT

**Comment by xige-16** [comment]
rerun ut

File: internal/proxy/task_delete.go [hunk 1]
```diff
@@ -171,6 +172,14 @@ func (dt *deleteTask) PreExecute(ctx context.Context) error {
 	dt.DeleteRequest.CollectionID = collID
 	dt.collectionID = collID
 
+	partitionKeyMode, err := isPartitionKeyMode(ctx, dt.CollectionName)
```
Comment (xiaofan-luan) [inline internal/proxy/task_delete.go:175]: how did delete handle partition ID?

**Comment by xige-16** [comment]
rerun ut

**Review by longjiquan — CHANGES_REQUESTED** [review]

File: internal/proxy/task_delete.go (hunk 1 above)
Comment (xige-16) [reply internal/proxy/task_delete.go:175]: milvus only support deleting on primary field

File: internal/proxy/expr_checker_test.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+	queryInfo := &planpb.QueryInfo{
+		Topk:         10,
+		MetricType:   "L2",
+		SearchParams: "",
+		RoundDecimal: -1,
+	}
+
+	t.Run("term", func(t *testing.T) {
+		exprStr := "partition_key_field in [7, 8] && partition_key_field not in [10, 20]"
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/proxy/expr_checker_test.go]: done

File: internal/proxy/task.go [hunk 3]
```diff
@@ -1042,6 +1105,14 @@ func (spt *showPartitionsTask) PreExecute(ctx context.Context) error {
 		return err
 	}
 
+	partitionKeyMode, err := isPartitionKeyMode(ctx, spt.CollectionName)
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/proxy/task.go]: done

File: internal/proxy/task.go [hunk 4]
```diff
@@ -155,6 +155,46 @@ func (cct *createCollectionTask) OnEnqueue() error {
...
+			if cct.GetNumPartitions() == 0 {
+				cct.NumPartitions = common.DefaultPartitionsWithPartitionKey
+			}
+
+			idx = i
+		}
+	}
+
+	if idx == -1 && cct.GetNumPartitions() != 0 {
+		return fmt.Errorf("partitions should be specified only on partition key field, field name = %s", cct.schema.Fields[idx].Name)
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/proxy/task.go]: done

**Pull request merged**